datasquare.go

```diff
@@ -187,29 +189,41 @@ func (ds *dataSquare) resetRoots() {
 	}
 }
 
-func (ds *dataSquare) computeRoots() {
-	var wg sync.WaitGroup
+func (ds *dataSquare) computeRoots() error {
+	g := new(errgroup.Group)
```

Comment:
New forces allocation on heap. There is no reason to do it here as it used within methods boundaries and should be kept on stack by var

Reply:
Does https://github.com/celestiaorg/rsmt2d/pull/157/commits/1956b1665b2669f945ecf23ce31d64d1211dd989 resolve your concern?

Reply:
Yes

Reply:
The quickest link I found: https://stackoverflow.com/a/10866871